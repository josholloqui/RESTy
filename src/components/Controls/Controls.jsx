import React from 'react';
import PropTypes from 'prop-types';

const Controls = ({ url, method, body, onChange, onSubmit }) => (
  <div>
    <form onSubmit={onSubmit} >
      <label htmlFor="url" >URL</label>
      <input
        id="url"
        type="text"
        name="url"
        value={url}
        onChange={onChange}
      />
      <section>
        <div>
          <label htmlFor="post" >POST</label>
          <input
            id="post"
            type="radio"
            name="method"
            value="POST"
            checked={method === 'POST'}
            onChange={onChange}
          />
          <label htmlFor="get" >GET</label>
          <input
            id="get"
            type="radio"
            name="method"
            value="GET"
            checked={method === 'GET'}
            onChange={onChange}
          />
          <label htmlFor="put" >PUT</label>
          <input
            id="put"
            type="radio"
            name="method"
            value="PUT"
            checked={method === 'PUT'}
            onChange={onChange}
          />
          <label htmlFor="patch" >PATCH</label>
          <input
            id="patch"
            type="radio"
            name="method"
            value="PATCH"
            checked={method === 'PATCH'}
            onChange={onChange}
          />
          <label htmlFor="delete" >DELETE</label>
          <input
            id="delete"
            type="radio"
            name="method"
            value="DELETE"
            checked={method === 'DELETE'}
            onChange={onChange}
          />
        </div>
        <button>Submit</button>
      </section>
      <textarea
        placeholder="Raw JSON Body"
        name="body"
        value={body}
        onChange={onChange}
      />
    </form>
  </div>
);

Controls.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Controls;
