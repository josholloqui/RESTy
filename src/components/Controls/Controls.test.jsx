import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Controls from './Controls';

describe('Controls component', () => {
  afterEach(() => cleanup());
  it('renders Controls component', () => {
    const { asFragment } = render(
      <Controls 
        url="https://anapioficeandfire.com/api/characters/583"
        method="GET"
        body=""
        onSubmit={() => {}}
        onChange={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
