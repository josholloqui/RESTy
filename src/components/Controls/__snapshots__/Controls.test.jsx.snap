// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders Controls component 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="url"
      >
        URL
      </label>
      <input
        id="url"
        name="url"
        type="text"
        value="https://anapioficeandfire.com/api/characters/583"
      />
      <section>
        <div>
          <label
            for="post"
          >
            POST
          </label>
          <input
            id="post"
            name="method"
            type="radio"
            value="POST"
          />
          <label
            for="get"
          >
            GET
          </label>
          <input
            checked=""
            id="get"
            name="method"
            type="radio"
            value="GET"
          />
          <label
            for="put"
          >
            PUT
          </label>
          <input
            id="put"
            name="method"
            type="radio"
            value="PUT"
          />
          <label
            for="patch"
          >
            PATCH
          </label>
          <input
            id="patch"
            name="method"
            type="radio"
            value="PATCH"
          />
          <label
            for="delete"
          >
            DELETE
          </label>
          <input
            id="delete"
            name="method"
            type="radio"
            value="DELETE"
          />
        </div>
        <button>
          Submit
        </button>
      </section>
      <textarea
        name="body"
        placeholder="Raw JSON Body"
      />
    </form>
  </div>
</DocumentFragment>
`;
